ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Stream.scala

```diff
@@ -441,12 +441,12 @@ private[h2] object H2Stream {
 
     private[H2Stream] def cancelWith(msg: String)(implicit F: Monad[F]): F[Unit] = {
       // Unsure of this, but also unsure about exposing custom throwable
-      val t = new CancellationException(msg)
-      writeBlock.complete(Left(t)) >>
-        request.complete(Left(t)) >>
-        response.complete(Left(t)) >>
-        readBuffer.send(Left(t)) >>
-        trailers.complete(Left(t)).void
+      val ex: Either[Throwable, Nothing] = Left(new CancellationException(msg))
+      writeBlock.complete(ex) >>
+        request.complete(ex) >>
+        response.complete(ex) >>
+        readBuffer.send(ex) >>
+        trailers.complete(ex).void
```

Comment:
Heh, thanks, I saw this one too the other day 😛

Reply:
If you want to optimize more, replace `>>` with `*>`, it will save 4 allocations.

Reply:
Hmmmm, I'm really not sure if this will save allocations. Could you please elaborate on this? 🤔

Reply:
Yes, actually my initial intuition is correct. Because `>>` is lazy it allocates a lambda.

```scala
def strict[F[_]](implicit F: Monad[F]) = F.unit *> F.unit

def lazy_[F[_]](implicit F: Monad[F]) = F.unit >> F.unit
```
decompiles to
```java
public <F> Object strict(Monad<F> F) {
    return package.all$.MODULE$.catsSyntaxApply(F.unit(), F).$times$greater(F.unit());
}

public <F> Object lazy_(Monad<F> F) {
    Object object = package.all$.MODULE$.catsSyntaxFlatMapOps(F.unit(), F);
    return FlatMapOps$.MODULE$.$greater$greater$extension(object, () -> Foo.lazy_$$anonfun$1(F), F);
                                                                  /* ^^^ lambda allocation ^^^ */
}

private static final Object lazy_$$anonfun$1(Monad F$1) {
    return F$1.unit();
}
```